pvacseq/lib/parse_output.py

```diff
@@ -15,63 +15,160 @@
 
 csv.field_size_limit(sys.maxsize)
 
+def parse_input_tsv_file(input_tsv_file):
+    tsv_reader = csv.DictReader(input_tsv_file, delimiter='\t')
+    tsv_entries = {}
+    for line in tsv_reader:
+        tsv_entries[line['index']] = line
+    return tsv_entries
+
 def min_match_count(peptide_length):
     return ceil(peptide_length / 2)
 
-def determine_consecutive_matches(mt_epitope_seq, wt_epitope_seq):
+def determine_consecutive_matches_from_left(mt_epitope_seq, wt_epitope_seq):
     consecutive_matches = 0
-    left_padding        = 0
-    #Count consecutive matches from the beginning of the epitope sequences
     for a, b in zip(mt_epitope_seq, wt_epitope_seq):
         if a == b:
             consecutive_matches += 1
-            left_padding        += 1
         else:
             break
-    #Count consecutive matches from the end of the epitope sequences
+    return consecutive_matches
+
+def determine_consecutive_matches_from_right(mt_epitope_seq, wt_epitope_seq):
+    consecutive_matches = 0
     for a, b in zip(reversed(mt_epitope_seq), reversed(wt_epitope_seq)):
         if a == b:
             consecutive_matches += 1
         else:
             break
-    return consecutive_matches, left_padding
+    return consecutive_matches
 
-def parse_input_tsv_file(input_tsv_file):
-    tsv_reader = csv.DictReader(input_tsv_file, delimiter='\t')
-    tsv_entries = {}
-    for line in tsv_reader:
-        tsv_entries[line['index']] = line
-    return tsv_entries
+def find_mutation_position_from_left(wt_epitope_seq, mt_epitope_seq):
+    for i,(wt_aa,mt_aa) in enumerate(zip(wt_epitope_seq,mt_epitope_seq)):
+        if wt_aa != mt_aa:
+            return i+1
+    return 0
 
 def match_wildtype_and_mutant_entries(iedb_results, wt_iedb_results):
-    for key, result in iedb_results.items():
+    for key in sorted(iedb_results.keys(), key = lambda x: int(x.split('|')[-1])):
+        result = iedb_results[key]
         (wt_iedb_result_key, mt_position) = key.split('|', 1)
+        wt_results = wt_iedb_results[wt_iedb_result_key]
         if result['variant_type'] == 'missense':
-            iedb_results[key]['wt_epitope_seq'] = wt_iedb_results[wt_iedb_result_key][mt_position]['wt_epitope_seq']
-            iedb_results[key]['wt_scores']      = wt_iedb_results[wt_iedb_result_key][mt_position]['wt_scores']
-        else:
-            wt_results        = wt_iedb_results[wt_iedb_result_key]
+            iedb_results[key]['wt_epitope_seq']    = wt_results[mt_position]['wt_epitope_seq']
```

Comment:
That's an interesting point. I'm not sure if `result` is a pointer or a copy. I think python handles these as pointers so it should work. I will test when I get back.